From b7d6ccea43312387c97487fb509caa07ec39d99d Mon Sep 17 00:00:00 2001
From: Apile Tyumre <apile@asantesolutions.co.za>
Date: Tue, 19 May 2026 12:51:42 +0200
Subject: [PATCH] Students can only see courses they are enrolled in and
 lecturers can only see courses that they are in charge of

---
 frontend/src/pages/Courses/Courses.vue | 36 ++++++++++++++++----------
 1 file changed, 22 insertions(+), 14 deletions(-)

diff --git a/frontend/src/pages/Courses/Courses.vue b/frontend/src/pages/Courses/Courses.vue
index 6e5ae9d..3b90da8 100644
--- a/frontend/src/pages/Courses/Courses.vue
+++ b/frontend/src/pages/Courses/Courses.vue
@@ -138,7 +138,7 @@ const currentCategory = ref(null)
 const title = ref('')
 const certification = ref(false)
 const filters = ref({})
-const currentTab = ref('live')
+const currentTab = user.data?.is_student ? ref('enrolled') : ref('live')
 const { brand } = sessionStore()
 const courseCount = ref(0)
 const router = useRouter()
@@ -167,6 +167,15 @@ const courses = createListResource({
 	cache: ['courses', user.data?.name],
 	pageLength: pageLength.value,
 	start: start.value,
+	transform(data) {
+		if(user.data?.is_instructor && (!user.data?.is_system_manager && !user.data?.is_moderator && !user.data?.is_evaluator)) {
+			return data.filter((element) => {
+				return element.instructors.some((instructor) => instructor.name === user.data?.name)
+			})
+		}
+
+		return data
+	},
 })
 
 const setCategories = (data) => {
@@ -313,25 +322,24 @@ watch(currentTab, () => {
 })
 
 const courseTabs = computed(() => {
-	let tabs = [
-		{
+	let tabs = []
+	if (
+		user.data?.is_moderator ||
+		user.data?.is_instructor ||
+		user.data?.is_evaluator
+	) {
+		tabs.push({
 			label: __('Live'),
 			value: 'live',
-		},
-		{
+		})
+		tabs.push({
 			label: __('New'),
 			value: 'new',
-		},
-		{
+		})
+		tabs.push({
 			label: __('Upcoming'),
 			value: 'upcoming',
-		},
-	]
-	if (
-		user.data?.is_moderator ||
-		user.data?.is_instructor ||
-		user.data?.is_evaluator
-	) {
+		})
 		tabs.push({ label: __('Created'), value: 'created' })
 		tabs.push({ label: __('Unpublished'), value: 'unpublished' })
 	} else if (user.data) {
-- 
2.43.0

